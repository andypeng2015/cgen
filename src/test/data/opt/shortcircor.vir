module foo

export function w $foo(l %a, l %i) {
@start:
  %k = sge.l %i, 3_l
  br %k, @join, @cont
@cont:
  %k = slt.l %i, 0_l
  jmp @join
@join:
  br %k, @bad, @ok
@ok:
  %s = mul.l %i, 4_l
  %a = add.l %a, %s
  %v = ld.w %a
  ret %v
@bad:
  ret 0xffffffff_w
}
