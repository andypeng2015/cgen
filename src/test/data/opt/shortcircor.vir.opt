module foo

export function w $foo(l %a, l %i) {
@2:
  %5 = sge.l %i, 0x3_l ; @17
  br %5, @12(0xffffffff_w), @4
@4:
  %6 = slt.l %i, 0x0_l ; @18
  br %6, @12(0xffffffff_w), @8
@8:
  %7 = lsl.l %i, 0x2_l ; @19
  %8 = add.l %a, %7 ; @20
  %v.1 = ld.w %8 ; @11
  jmp @12(%v.1)
@12(%0):
  ret %0
}
