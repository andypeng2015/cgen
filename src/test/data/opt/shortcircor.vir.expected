module foo

export function w $foo(l %a, l %i) {
@2:
  %1 = sge.l %i, 0x3_l ; @13
  br %1, @12(0xffffffff_w), @4
@4:
  %2 = slt.l %i, 0x0_l ; @14
  br %2, @12(0xffffffff_w), @8
@8:
  %3 = lsl.l %i, 0x2_l ; @15
  %4 = add.l %a, %3 ; @16
  %v.1 = ld.w %4 ; @11
  jmp @12(%v.1)
@12(%0):
  ret %0
}
